### DISCUSSION TITLE
Does Cvxpy exploits  sparsity of the optimization problem?

### DISCUSSION THREAD

**Discussion by pqrz** [cvxpy/cvxpy#1895] [Q&A] [answered]
What happens when a (sparse) `csr` matrix / array is submitted to Cvxpy.

Does it exploit the sparsity Information about the matrix or does everything gets converted to standard matrix / array?

Note: We are using Gurobi solver in Cvxpy

**Comment by SteveDiamond** [comment, marked as answer]
It retains the sparsity of the data and passes it down to the solver as a sparse matrix.
